service/app.py

```diff
@@ -0,0 +1,103 @@
+import os
+import subprocess
+import uuid
+
+from flask import Flask
+from flask_restful import Resource, Api, reqparse
+
+import werkzeug
+
+# Set up the flask app and API
+app = Flask(__name__)
+api = Api(app)
+
+# Declare two global variables that will hold state.
+WORKFLOW_TESTS = {}
+ENVIRONMENTS = {}
+
+# Functions for running the test
+def submit_test(environment_image, workflow_source, workflow_inputs, workflow_dependencies):
+
+    docker_cmd = ["docker", "run", "-v /io:/io", "-v /var/run/docker.sock:/var/run/docker.sock", environment_image]
+    docker_cmd.extend(["--workflowSource", workflow_source])
+    docker_cmd.extend(["--workflowInputs", workflow_inputs])
+    if workflow_dependencies:
+        docker_cmd.extend(["--workflowDependencies", workflow_dependencies])
+
+    proc = subprocess.Popen(docker_cmd)
+    proc.communicate()
+
+
+    if proc.returncode == 0:
+        workflow_test = {"status": "Succeeded"}
+    else:
+        workflow_test = {"status": "Failed"}
+    workflow_test["command"] = ' '.join(docker_cmd)
+
+    return workflow_test
+
+class WorkflowTest(Resource):
+    def get(self):
+        parser = reqparse.RequestParser()
+        parser.add_argument("workflowTestId")
+        args = parser.parse_args()
+        return WORKFLOW_TESTS[args.WorkflowTestId]
+
+    def post(self):
+        parser = reqparse.RequestParser()
+        parser.add_argument("workflowSource")
+        parser.add_argument("workflowInputs")
+        parser.add_argument("workflowDependencies",
+                            type=werkzeug.datastructures.FileStorage,
+                            location='files')
+        args = parser.parse_args()
+
+        # Create a unique id for this workflow test
+        workflow_test_id = str(uuid.uuid4())
+
+        # Store the submitted WDLs and inputs.
+        os.makedirs(os.path.join(os.sep, 'io', workflow_test_id))
+        wdl_path = os.path.join(os.sep, "io", workflow_test_id, "source.wdl")
+        inputs_path = os.path.join(os.sep, "io", workflow_test_id, "inputs.json")
+        deps_path = os.path.join(os.sep, "io", workflow_test_id, "dependencies.zip")
+
+        with open(wdl_path, 'w') as wdl:
```

Comment:
Does it make sense to store these as temp files, or is there a reason to persist them after you've posted the workflow?

Reply:
They have to go into `/io/...` because of docker-within-docker issues for the local cromwell environment. It's also kind of nice to keep them around for debugging, though I think I'll note that there will have to be a cleanup step at some point.